SDK.md

```diff
@@ -0,0 +1,281 @@
+# CloudEvents SDK
+
+The intent of this document to describe bare minimum set of 
+requirements for a new SDKs.
+
+## Status of this document
+
+Since [CloudEvent spec](spec.md) still considered as a working draft this 
+document also suppose to be considered as a working draft.
+
+## Contribution acceptance
+
+Being an open source community CloudEvents team is open for a new members 
+as well open to their contribution.
+In order to ensure that an SDK is going to be supported and maintained 
+CloudEvents community would like to ensured that:
+
+ - a person (developer, committer) is going to become a maintainer
+ - a person commits to support ongoing changes to the [CloudEvent spec](spec.md)
+
+## Officially maintained software development kits (SDK)
+
+Software Development Kit (SDK) is a set of helpers designed and implemented to 
+enhance and speed up a CloudEvents integration.
+As part of community efforts CloudEvents team committed to support and maintain 
+the following SDKs:
+
+ - [Go SDK](https://github.com/cloudevents/sdk-go)
+ - [Java SDK](https://github.com/cloudevents/sdk-java)
+ - [Python SDK](https://github.com/cloudevents/sdk-python)
+ - [CSharp](https://github.com/cloudevents/sdk-csharp)
+ - [JavaScript SDK](https://github.com/cloudevents/sdk-javascript)
+
+## SDK contribution
+
+That's said, being an open source community CloudEvents is all about building 
+open for contribution.
+In order to keep the development/user experience the same (as much as possible) 
+CloudEvents team agreed to establish minimum requirements for a new SDK.
+
+### Technical requirements
+
+Supports CloudEvents spec milestones and ongoing development version.
+HTTP transport support (both structured and binary).
+Widely used programming language version.
+
+
+### Preferable API signature guidelines
+
+In order to remain developer/user UX the same among existing officially supported 
+SDKs CloudEvents team asks maintainers to align with the following API signatures.
+Please consider the following code as pseudo-code.
+
+#### Event object constructor API
+
+Event build considered to be an event constructor:
+```
+    v01.Event()
+```
+
+#### Event object setters API 
+
+This particular code sample represents bare minimum number of setters:
+```
+    v01.Event().
+    SetContentType("application/json").
+    SetData('{"name":"john"}').
+    SetEventID("my-id").
+    SetSource("from-galaxy-far-far-away").
+    SetEventTime("tomorrow").
+    SetEventType("cloudevent.greet.you")
+```
+
+Content type setter represents an event MIME content type setter:
+```
+    SetContentType(content_type string)
+```
+
+Data setter represents an event data setter:
+```
+    SetData(event_data serializable)
+```
+
+ID setter represents an event ID setter:
+```
+    SetEventID(id string)
+```
+
+Source setter represents an event source setter:
+```
+    SetSource(source URL)
+```
+
+Time setter represents event emit time setter:
+```
+    SetEventTime(time RFC3339)
```

Comment:
let's use something like `SetTime`, since that actually matches the 0.2 bits ...

Reply:
I'd rather leave this as it is, the small bonus here is the verbosity, once 0.1 will be deprecated, we can switch to whatever that is.